Automattic/Gravatar-SDK-iOS
# Simplify OpenAPI Generator

Closes #

### Description

This simplifies our OpenAPI Generator setup, and borrows from the work done in #486:
- Simplifies the `generate` command:
   - Calls: `docker run --rm openapitools/openapi-generator-cli:"v7.5.0" generate`
      - It fetches the **_official docker container_** instead of using the locally fetched generator in a custom docker container
      - The `install-and-generate` Makefile task has been removed, as it is no longer required
   - Generates all file types by default (instead of just `models`)
      - Uses `.openapi-generator-ignore` to selectively ignore which file/folders should be created
   - Passes new options to `--additional-options`
- Moves all `openapi-generate` activity into the `opeapi` directory
   - Files are generated here, and are copied elsewhere as needed
      - The `swift5` generator supports outputing source files as a Swift Package.  While we aren't using it as a Swift Package, using that folder structure feels more natural.  So the generated source files are in `Sources`.  They are then rsynced to their final destination

### Note
After this PR merges, we will want to delete the `openapi-generator` directory in the root of the repo. 

This PR no longer requires or uses the `openapi-generator` directory.  That folder is currently ignored by git.  But since we don't need this folder any more, this PR stops ignoring that folder.

### Testing Steps

1. Make sure Docker is running locally
2. Run `make generate`
- OBSERVE:
   - [ ] Docker downloads the proper image (`7.5.0`) (only occurs the first time):
```shell
Unable to find image 'openapitools/openapi-generator-cli:v7.5.0' locally
v7.5.0: Pulling from openapitools/openapi-generator-cli
a5319f8e5f3f: Pull complete
d3aab5129936: Pull complete
b47927223c23: Pull complete
b2a3e41ae21b: Pull complete
914200ed739a: Pull complete
2b2d8033142b: Pull complete
505e99177f70: Pull complete
Digest: sha256:cdf11948948de9c21c6035de47dd5fc73c1651c8ba2ea0a4b86a527608ef52a9
Status: Downloaded newer image for openapitools/openapi-generator-cli:v7.5.0
```
   - [ ] The only changes to source files in `Sources/Gravatar/OpenApi/Generated` should be the usual changes to access `package --> public` levels, which we are currently correcting by hand.
```diff
- package struct Avatar: Codable, Hashable, Sendable {
-    package enum Rating: String, Codable, CaseIterable, Sendable {
+ public struct Avatar: Codable, Hashable, Sendable {
+    public enum Rating: String, Codable, CaseIterable, Sendable {
```
